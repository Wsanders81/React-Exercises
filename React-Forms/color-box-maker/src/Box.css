.Box-container {
    text-align: center;
    margin: auto;
     
}

.Box {
    margin: auto; 
    margin-top: 1rem; 
}