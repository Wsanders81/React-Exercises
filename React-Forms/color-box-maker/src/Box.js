import React from "react";
import "./Box.css"
const Box = ({ width = 5, height = 5, color = '#322a2a' , deleteBox, id}) => {
const remove = () => {deleteBox(id)}
    return (
        <div className="Box-container">
        <div className="Box" 
        style = {{
            backgroundColor : `${color}`,
            width: `${width}rem`,
            height: `${height}rem`
            }}>
        
        </div>
        <div>
            <button onClick={remove}>REMOVE</button>
        </div>
        </div>
    )
}

export default Box; 