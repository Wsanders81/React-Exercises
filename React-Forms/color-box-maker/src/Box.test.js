import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

it('renders without crashing', () => {
    render(<Box width={5} height={5} color="blue"/>)
})

it('matches snapshot', () => {
    const { asFragment } = render(<Box width={5} height={5} color="blue"/>)
    expect(asFragment()).toMatchSnapshot()
})