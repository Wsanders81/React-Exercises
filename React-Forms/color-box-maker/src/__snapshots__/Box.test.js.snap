// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box-container"
  >
    <div
      class="Box"
      style="background-color: blue; width: 5rem; height: 5rem;"
    />
    <div>
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
